contracts/RecruitmentV2.sol

```diff
@@ -0,0 +1,434 @@
+        require(_candidateEmail.length > 0, "Invalid candidate email");
+        require(_referralCode.length > 0, "Invalid referral code");
+        require(jobs[jobId].company != address(0), "Job does not exist");
+        require(jobs[jobId].isOpen, "Job is not open");
+        uint256 referralId = nextReferralId++;
+
+        Referral memory newReferral = Referral(
+            msg.sender,
+            jobId,
+            _candidateEmail,
+            _referralCode,
+            block.timestamp,
+            referralId,
+            false,
+            false
+        );
+
+        referrals[referralId] = newReferral;
+        referrerToReferrals[msg.sender].push(referralId);
+        jobIdRefferals[jobId].push(referralId);
+        companyReferred[jobs[jobId].company][msg.sender] = true;
+        emit ReferralMade(msg.sender, referralId);
+    }
+
+    /// Confirm Refferal by the candidate
+    /// @param _referralId referral id of the referral
+    /// @param _referralCode referral code of the referrer
+    /// @param _candidateEmail hash of candidate email in bytes32
+    function confirmReferral(
+        uint256 _referralId,
+        bytes32 _referralCode,
+        bytes32 _candidateEmail
+    ) external {
+        uint256 _jobId = referrals[_referralId].jobId;
+        require(
+            referrals[_referralId].candidateEmail == _candidateEmail,
+            "Not a candidate"
+        );
+        require(
+            referrals[_referralId].isConfirmed == false,
+            "Already confirmed"
+        );
+        require(referrals[_referralId].jobId == _jobId, "Invalid job id");
+        require(
+            referrals[_referralId].referralCode == _referralCode,
+            "Invalid referral code"
+        );
+        require(jobs[_jobId].isOpen, "Job is not open");
+        require(jobs[_jobId].company != address(0), "Job does not exist");
+        require(
+            referrals[_referralId].timeCreatead + 2 weeks > block.timestamp,
+            "Referral expired"
+        );
+        if (candidates[_candidateEmail].candidateAddress == address(0)) {
+            candidateAddressToEmail[msg.sender] = _candidateEmail;
+            candidates[_candidateEmail] = Candidate(
+                msg.sender,
+                _candidateEmail,
+                0
+            );
+        }
+
+        referrals[_referralId].isConfirmed = true;
+        emit ReferralConfirm(_referralId, _jobId, _candidateEmail);
+    }
+
+    /// Hire a candidate
+    /// @param referralId referral id of the referral
+    function hireCandidate(
+        uint256 referralId
+    ) external onlyRegisteredCompany onlyOpenJob(referrals[referralId].jobId) {
+        Referral storage referral = referrals[referralId];
+        Job storage job = jobs[referral.jobId];
+
+        require(msg.sender == job.company, "Only the company can hire");
+        require(!referral.isHired, "Candidate already hired");
+
+        referral.isHired = true;
+        job.isOpen = false;
```

Comment:
```suggestion
job.vacancyCount--
if(job.vacancyCount == 0){
        job.isOpen = false;
}
```

Reply:
related to the previous comment, we need to have a consensus about handling the vacancies.